net.solarnetwork.node.weather.ibm.wc.test/src/net/solarnetwork/node/weather/ibm/wc/test/WCClientTests.java

```diff
@@ -0,0 +1,122 @@
+package net.solarnetwork.node.weather.ibm.wc.test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import java.math.BigDecimal;
+import java.text.SimpleDateFormat;
+import java.util.Collection;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import net.solarnetwork.node.domain.GeneralDayDatum;
+import net.solarnetwork.node.weather.ibm.wc.BasicWCClient;
+import net.solarnetwork.node.weather.ibm.wc.WCClient;
+import net.solarnetwork.node.weather.ibm.wc.WCDayDatumDataSource;
+import net.solarnetwork.node.weather.ibm.wc.WCHourlyDatum;
+import net.solarnetwork.node.weather.ibm.wc.WCHourlyDatumDataSource;
+
+public class WCClientTests extends AbstractHttpClientTests{
+	
+	private BasicWCClient c;
+	private static final String RESOURCE_NAME = "7day_daily.json";
+
+	private final Logger log = LoggerFactory.getLogger(getClass());
+
+	@Before
+	public void additionalSetup() {
+		c = new BasicWCClient();
+		c.setBaseUrl(getHttpServerBaseUrl());
+		c.setHourlyForecastUrl("%s_hourly.json");
+		c.setDailyForecastUrl("%s_daily.json");
+		c.setObjectMapper(new ObjectMapper());
+	}
+
+	@Test
+	public void readHourlyForecast() {
+
+		TestHttpHandler handler = new TestHttpHandler() {
+			@Override
+			protected boolean handleInternal(HttpServletRequest request, HttpServletResponse response)
+					throws Exception {
+				assertEquals("GET", request.getMethod());
+				respondWithResource(response, "2day_hourly.json");
+				response.flushBuffer();
+				return true;
+			}
+		};
+		getHttpServer().addHandler(handler);
+		Collection<WCHourlyDatum> datum = c.readHourlyForecast("", "", "");
```

Comment:
This test could be improved by passing in actual values here instead of empty strings. Then up after line 45 you'd verify that the requested path was the expected path. This then verifies the client encodes the URL as expected, which currently is not verified.